src/bun.js/event_loop.zig

```diff
@@ -1275,6 +1277,7 @@ pub const EventLoop = struct {
 
     pub fn tickImmediateTasks(this: *EventLoop, virtual_machine: *VirtualMachine) void {
         _ = this.tickQueueWithCount(virtual_machine, "immediate_tasks");
+        JSC.napi.Finalizer.drain(&this.napi_finalizer_queue);
```

Comment:
This should not be here. Instead, it should enqueue a task to drain the napi finalizers when there are any finalizers to drain.